storage/series/buffer.go

```diff
@@ -240,6 +240,10 @@ func bucketDrainAndReset(now time.Time, b *dbBuffer, idx int, start time.Time) i
 			mergedOutOfOrderBlocks = 1
 		}
 		if merged.block.Len() > 0 {
+			// If this block was read mark it as such
+			if lastRead := b.buckets[idx].lastRead(); !lastRead.IsZero() {
+				merged.block.SetLastReadTime(lastRead)
```

Comment:
What happens if the block already has a last read time? Do we want to take the latter of the two read times?